### PR TITLE
fix(cmd/influxd): Add logging work-around for absolute file paths on Windows

### LINKED ISSUE #21534: Migration 1.x to 2.0.x fails on Windows : couldn't open sink
Windows 7 with powershell 7
Migrating from influxdb 1.7.6 to 2.0.6 using command .\influxd.exe upgrade
Working directory is "C:\influxdb"

-> An error occurs : couldn't open sink "C:\\Users\\MY_CURRENT_USER\\upgrade.log": no sink found for scheme "c"

I googled and found that this may need some tweaking in the migration code to make the sink compatible with Windows : https://github.com/uber-go/zap/issues/621

Did someone manage to uprade under windows?

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #21534

I tested this manually, waiting on the original reporter's feedback to confirm that the fix works for them too.

### CODE REVIEW

File: cmd/influxd/upgrade/logging_windows.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func buildLogger(options *logOptions, verbose bool) (*zap.Logger, error) {
+	config := zap.NewProductionConfig()
+
+	config.Level = zap.NewAtomicLevelAt(options.logLevel)
+	if verbose {
+		config.Level.SetLevel(zap.DebugLevel)
+	}
+
+	logPath := "winfile:///" + filepath.Abs(options.logPath)
```
Comment (davidby-influx) [inline cmd/influxd/upgrade/logging_windows.go]: I feel like buildLogger could be unified into one procedure, with a little dependency injection to fix up logPath, like a `generateOsDependentLogPath()` or something (but better-named).

**Review by davidby-influx — APPROVED** [review]
Code looks correct, and I offer a non-blocking refactoring suggestion to consider.

File: cmd/influxd/upgrade/logging_windows.go (hunk 1 above)
Comment (danxmoran) [reply cmd/influxd/upgrade/logging_windows.go]: Nice idea, will do 👍

**Review by davidby-influx — APPROVED** [review]
Very clean!

**Pull request merged**